src/commands/server/update.ts

```diff
@@ -110,40 +120,114 @@ export default class Update extends Command {
+      const newOperatorImage = `${ctx.newCheOperatorImage}:${ctx.newCheOperatorTag}`
+      cli.info(`Existed Eclipse Che operator: ${existedOperatorImage}.`)
+      cli.info(`New Eclipse Che operator    : ${newOperatorImage}.`)
 
-        if (flags['che-operator-image'] !== DEFAULT_CHE_OPERATOR_IMAGE) {
-          cli.warn(`This command updates Eclipse Che to ${getImageTag(DEFAULT_CHE_OPERATOR_IMAGE)} version, but custom operator image is specified.`)
-          cli.warn('Make sure that the new version of the Eclipse Che is corresponding to the version of the tool you use.')
-          cli.warn('Consider using \'chectl update [stable|next]\' to update to the latest version of chectl.')
-        }
+      const defaultOperatorImageTag = getImageTag(DEFAULT_CHE_OPERATOR_IMAGE)
+      const chectlChannel = defaultOperatorImageTag === 'nightly' ? 'next' : 'stable'
+      const currentChectlVersion = getCurrentChectlVersion()
+      const latestChectlVersion = await getLatestChectlVersion(chectlChannel)
+      const projectName = getCurrentChectlName()
 
-        const cheCluster = await kubeHelper.getCheCluster(flags.chenamespace)
-        if (cheCluster.spec.server.cheImage
-          || cheCluster.spec.server.cheImageTag
-          || cheCluster.spec.server.devfileRegistryImage
-          || cheCluster.spec.database.postgresImage
-          || cheCluster.spec.server.pluginRegistryImage
-          || cheCluster.spec.auth.identityProviderImage) {
-          cli.warn(`In order to update Eclipse Che the images defined in the '${cheCluster.metadata.name}'
-            Custom Resource of the namespace '${flags.chenamespace}' will be cleaned up:`)
-          cheCluster.spec.server.cheImageTag && cli.warn(`Eclipse Che server image tag [${cheCluster.spec.server.cheImageTag}]`)
-          cheCluster.spec.server.cheImage && cli.warn(`Eclipse Che server [${cheCluster.spec.server.cheImage}]`)
-          cheCluster.spec.database.postgresImage && cli.warn(`Database [${cheCluster.spec.database.postgresImage}]`)
-          cheCluster.spec.server.devfileRegistryImage && cli.warn(`Devfile registry [${cheCluster.spec.server.devfileRegistryImage}]`)
-          cheCluster.spec.server.pluginRegistryImage && cli.warn(`Plugin registry [${cheCluster.spec.server.pluginRegistryImage}]`)
-          cheCluster.spec.auth.identityProviderImage && cli.warn(`Identity provider [${cheCluster.spec.auth.identityProviderImage}]`)
+      // the same version is already installed
+      if (newOperatorImage === existedOperatorImage) {
+        if (projectName === 'chectl' && latestChectlVersion) {
+          // suggest update chectl first
+          if (currentChectlVersion !== latestChectlVersion) {
+            cli.warn(`It is not possible to update Eclipse Che to a newer version
+using the current '${currentChectlVersion}' version of chectl. Please, update 'chectl'
+to a newer version '${latestChectlVersion}' with the command 'chectl update ${chectlChannel}'
+and then try again.`)
```

Comment:
Maybe move into a function with args that returns the message? Then block below will not be duplicated.

Reply:
It is slightly different.